flexmock.py

```diff
@@ -847,7 +847,11 @@ def _update_method(self, expectation, name):
             if method_type in SPECIAL_METHODS:
                 expectation.original_function = getattr(obj, name)
             expectation.method_type = method_type
-        if not _isclass(obj) or expectation.method_type in SPECIAL_METHODS:
+        if (
+            not _isclass(obj)
+            or expectation.method_type in SPECIAL_METHODS
+            or name == '__new__'
```

Comment:
What would happen if we added __new__ to the special methods?

Reply:
Special methods doesn't actually contain method names but types:
```python
SPECIAL_METHODS = (classmethod, staticmethod)
```